packages/nx/src/command-line/yargs-utils/shared-options.ts

```diff
@@ -214,6 +216,8 @@ export function withRunManyOptions<T>(
       'strip-dashed': true,
       'unknown-options-as-args': true,
       'populate--': true,
+      'parse-numbers': false,
+      'parse-positional-numbers': false,
```

Comment:
I feel like other stuff will break with this... what about like `--maxWarnings 5` or `--parallel 8`? :eyes: 

Did you check those?

Reply:
i test it out:
- for yargs option with type number (e.g. `maxParallel`), it will be converted as number, `'parse-numbers': false,` will be applied to for non-nx override options (e.g. `_: [ '--max-warnings', '4', '--tag', '81919e4' ],`
- for parallel, the yargs option is string, however, there is a function `readParallelFromArgsAndEnv` that will do the parsing (https://github.com/nrwl/nx/blob/add4183d0a9acebd4ba7e28674a4597281c55828/packages/nx/src/utils/command-line-utils.ts#L189)
- for the non nx args that is going to pass to executor, there is additional parsing in function createOverrides (https://github.com/nrwl/nx/blob/add4183d0a9acebd4ba7e28674a4597281c55828/packages/nx/src/utils/command-line-utils.ts#L42) which does not have `'parse-numbers': false,`, so it will parse the args like `maxWarnings` to number

in summary, this `'parse-numbers': false,` will only be applied to non-nx override options, but not when it got passed to executor